vertical-pod-autoscaler/pkg/admission-controller/main.go

```diff
@@ -127,13 +132,16 @@ func main() {
 	})
 	clientset := getClient()
 	server := &http.Server{
-		Addr:      fmt.Sprintf(":%d", *port),
-		TLSConfig: configTLS(clientset, certs.serverCert, certs.serverKey),
+		Addr: fmt.Sprintf(":%d", *port),
 	}
+	// this will check if there are new certs before every tls handshake
+	t := &tls.Config{GetCertificate: kpr.GetCertificateFunc()}
```

Comment:
server.TLSConfig = &tls.Config{GetCertificate: kpr.GetCertificateFunc()}

you can also do it above:
server := &http.Server{
		Addr:      fmt.Sprintf(":%d", *port),			
		TLSConfig: &tls.Config{GetCertificate: kpr.GetCertificateFunc()},	
}

One more thing is that the KeypairReloader could have a GetTLSConfig function and then we could simply have:

server := &http.Server{
		Addr:      fmt.Sprintf(":%d", *port),			
		TLSConfig: kpr.GetTLSConfig(),
}